msgs/src/header.cpp

```diff
@@ -22,11 +22,12 @@ Header::Header(const void* data)
 Header::Header(const Header& h) : Header{h.seq_n_, h.frame_id_, h.timestamp_} {}
 
 Header::Header(Header&& other) noexcept
-  : seq_n_{std::move(other.seq_n_)}, frame_id_(std::move(other.frame_id_)), timestamp_{std::move(other.timestamp_)} {}
+  : seq_n_{other.seq_n_}, frame_id_(std::move(other.frame_id_)), timestamp_{other.timestamp_} {}
```

Comment:
i think there should be some locking here..

Reply:
This made me realize that some locking is needed in all the copy/move constructors of simple_msgs...

I'll add private constructors that take the object to be copied/move and a lock_guard, public copy/move constructors will call those passing over the object (other) and a lock_guard that uses the other.mutex_.